Flaky test `test_deterministic_of_observed_modified_interface`

```
FAILED pymc3/tests/test_sampling.py::TestSamplePPC::test_deterministic_of_observed_modified_interface
```

This test failed here: https://github.com/pymc-devs/pymc3/runs/2423911879

And I think I have seen it before as well. We should check if it's a newly introduced bug or just needs seeding.

--- a/pymc3/tests/test_sampling.py
+++ b/pymc3/tests/test_sampling.py
@@ -685,10 +685,10 @@ class TestSamplePPC(SeededTest):
         meas_in_1 = pm.aesaraf.floatX(2 + 4 * np.random.randn(100))
         meas_in_2 = pm.aesaraf.floatX(5 + 4 * np.random.randn(100))
         with pm.Model() as model:
-            mu_in_1 = pm.Normal("mu_in_1", 0, 1)
-            sigma_in_1 = pm.HalfNormal("sd_in_1", 1)
-            mu_in_2 = pm.Normal("mu_in_2", 0, 1)
-            sigma_in_2 = pm.HalfNormal("sd__in_2", 1)
+            mu_in_1 = pm.Normal("mu_in_1", 0, 1, testval=0)
+            sigma_in_1 = pm.HalfNormal("sd_in_1", 1, testval=1)
+            mu_in_2 = pm.Normal("mu_in_2", 0, 1, testval=0)
+            sigma_in_2 = pm.HalfNormal("sd__in_2", 1, testval=1)
 
             in_1 = pm.Normal("in_1", mu_in_1, sigma_in_1, observed=meas_in_1)
             in_2 = pm.Normal("in_2", mu_in_2, sigma_in_2, observed=meas_in_2)

